executor/src/witgen/machines/block_machine.rs

```diff
@@ -257,24 +257,6 @@ impl<T: FieldElement> BlockMachine<T> {
     ) -> EvalResult<'b, T> {
         log::trace!("Start processing block machine");
 
-        // First check if we already store the value.
```

Comment:
@chriseth Can you explain the purpose of this block?

My understanding is:
- This intends to catch the case that we already successfully processed the lookup. But in that case, the identity will be marked as complete in the Generator's main loop and won't be processed again anyway.
- This statement will *always* be true if we're in the `AssumeZero` strategy (all left values are constant and `self.rows() > 0` is always true since we add a dummy block).
  - In that case, the `assert!(value.constraints.is_empty())` will fail.

I think the only reason it would get to the `AssumeZero` strategy is because the system is under-constrained: All inputs to the block machine should have been figured out by the main SM, and if all inputs are given, the block machine should have figured out all outputs.

So if my understanding is right, it should be fine to remove that block and fail hard if we ever get to the `AssumeZero` strategy. Does that make sense?

Reply:
Yes, your analysis sounds correct. This code is from a time where we did not have the complete marker.